solr/core/src/java/org/apache/solr/search/SolrReturnFields.java

```diff
@@ -547,6 +547,12 @@ private void addField(
       String disp = (key == null) ? field : key;
       augmenters.addTransformer(new MatchScoreAugmenter(disp));
       scoreDependentFields.put(disp, disp.equals(MATCH_SCORE) ? "" : MATCH_SCORE);
+    } else if (key != null && isPseudoField) {
+      // SOLR-15030: a pseudo-field based on the function query may need scores,
+      // so we consider all pseudo-fields as potentially requiring scores.
+      // At the same time, we don't set _wantScore = true because the field
+      // list must explicitly include 'score' field to enable scores
```

Comment:
I'm confused or challenge what you say in this sentence.  A DocTransformer could want the score.  Granted, a DocTransformer doesn't have a `wantsScore` / `needsScore` so... :-/

Reply:
What I thought was that for the query with `fl=id,a,b,s:sum(x,y)` - even though it has a pseudo-field, that alone is not enough to conclude that the query requires scores.